frontend/src/constants/data.tsx

```diff
@@ -8,26 +11,48 @@ export const navLinks:NavLink[] = [
     {name:'Projects'},
 ];
 
+interface LinkProps {
+    icon: React.ElementType;
+    url: string;
+}
+
+export const Link: React.FC<LinkProps> = ({ icon: Icon, url }) => {
```

Comment:
We can move this into the component itself since this is a component instead of a constant.